sklearn/utils/estimator_checks.py

```diff
@@ -1105,7 +1126,10 @@ def check_array_api_input(
             )
         else:
             assert attribute.shape == est_xp_param_np.shape
-            assert attribute.dtype == est_xp_param_np.dtype
+            if _check_attributes_dtype_with_array_api(
```

Comment:
Do we need to check this like this? Having to maintain a list of estimators (and namespaces) inside the function seems tedious. It seems like we could check that the attribute's dtype is either equal to the dtype of the numpy based estimator (this is what we check now) or the max precision float dtype of the namespace/device (if that isn't float64).

If that is the right logic, would it be nicer to use that instead of maintaining a separate function and skip the check?

Reply:
I think we can certainly try that. We would need a specific check for mps generically then which should work. I separated this out because I thought not all estimators might need this kind of a logic.